packages/expo/src/hooks/useEvent.ts

```diff
@@ -0,0 +1,75 @@
+import type { EventEmitter } from 'expo-modules-core/types';
+import { useEffect, useState } from 'react';
+
+type AnyEventListener = (...args: any[]) => any;
+
+/**
+ * Type helper that infers the event name from the emitter's events map.
+ */
+type InferEventName<TEventsMap> =
+  TEventsMap extends Record<infer TEventName extends keyof TEventsMap, AnyEventListener>
+    ? TEventName
+    : never;
+
+/**
+ * Type helper that infers the event listener from the emitter's events map.
+ */
+type InferEventListener<TEventsMap, TEventName extends keyof TEventsMap> =
+  TEventsMap extends Record<TEventName, infer TEventListener extends AnyEventListener>
+    ? TEventListener
+    : never;
+
+/**
+ * Type helper that infers the first parameter of the event listener.
+ */
+type InferEventParameter<
+  TEventListener extends AnyEventListener,
+  TInitialValue,
+> = TInitialValue extends Parameters<TEventListener>[0]
+  ? Parameters<TEventListener>[0]
+  : Parameters<TEventListener>[0] | TInitialValue | null;
+
+/**
+ * React hook that listens to events emitted by the given object. The returned value is an array of event parameters
+ * that get updated whenever a new event is dispatched.
+ * @param emitter An object that emits events, e.g. a native module or shared object or an instance of [`EventEmitter`](#eventemitter).
```

Comment:
```suggestion
 * @param emitter An object that emits events. For example, a native module or shared object or an instance of [`EventEmitter`](#eventemitter).
```

Let's avoid using [Latin abbreviations as per our writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#abbreviations)